### ISSUE TITLE
AutoFactory generating the @Generated annotation on Android

### ISSUE THREAD

**Issue by AngusMorton** [google/auto#630]
AutoFactories are generated with an `@Generated` annotation on Android, this was an issue in Dagger https://github.com/google/dagger/issues/95. 

AutoValue behaves correctly in that it doesn't emit the `@Generated` annotation in Android where the annotation doesn't exist.

I'm running AutoFactory as an `annotationProcessor`, and have copied the annotations (`Provided`, and `AutoFactory`) into my project.

```
annotationProcessor 'com.google.auto.factory:auto-factory:1.0-beta5'
```

**Comment by jamienourish** [comment]
I had the same problem, adding 
  implementation 'org.glassfish:javax.annotation:3.1.1' 
enabled my project to compile

**Comment by AngusMorton** [comment]
Yeah, the workaround is to add the `@Generated` annotation from somewhere else, I used `javax.annotation:jsr250-api:1.0`, but Dagger and Autovalue both work without the `@Generated` annotation so AutoFactory should probably work without it too.

**Comment by dbarwacz** [comment]
Dagger adds @Generated for me in 2.14.1, maybe it looks whether this annotation is included and only then uses it.
